### PR TITLE
Updated travis yml to properly test different symfony versions

### PR DESCRIPTION
**Pull request by Nyholm**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| Documentation   | 
| License         | MIT

#### What's in this PR?

This will avoid us requiring symfony/symfony

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -33,15 +33,14 @@
...
-        "phpunit/phpunit": "^4.8.35 || ^5.7 || ^6.1",
         "phpunit/php-token-stream": "^1.1.8",
         "php-http/curl-client": "^1.0",
         "php-http/socket-client": "^1.0",
         "php-http/guzzle6-adapter": "^1.1.1",
         "php-http/react-adapter": "^0.2.1",
         "php-http/buzz-adapter": "^0.3",
         "php-http/mock-client": "^1.0",
-        "symfony/phpunit-bridge": "^3.2",
+        "symfony/phpunit-bridge": "^3.4@rc || ^4.0",
```
Comment (Nyholm) [inline composer.json]: 3.4 will allow PHPunit6.

File: .travis.yml [hunk 2]
```diff
@@ -24,26 +18,47 @@ branches:
...
         - php: 7.0
-          env: SYMFONY_VERSION=2.8.*
+        - php: 7.1
+        - php: 7.2
+
+          # Test LTS versions
+        - php: 5.5
+          env: DEPENDENCIES="symfony/lts:^2"
+        - php: 5.5
+          env: DEPENDENCIES="symfony/lts:^3"
```
Comment (fbourigault) [inline .travis.yml]: This allows some dependencies to be installed in their 2.x version. Is that wanted?

File: .travis.yml [hunk 3]
```diff
@@ -24,26 +18,47 @@ branches:
...
+
+          # Latest commit to master
+        - php: 7.1
+          env: DEPENDENCIES="dev"
+
+    allow_failures:
+          # Latest beta is allowed to fail.
```
Comment (fbourigault) [inline .travis.yml]: This is not `beta` but `dev-master`.

File: .travis.yml [hunk 4]
```diff
@@ -24,26 +18,47 @@ branches:
 matrix:
     fast_finish: true
     include:
+          # Minimum supported PHP and Symfony version
```
Comment (fbourigault) [inline .travis.yml]: Here PHP is the highest available version

File: composer.json (hunk 1 above)
Comment (xabbuh) [reply composer.json]: that should be possible with older versions of the bridge too

File: .travis.yml (hunk 2 above)
Comment (Nyholm) [reply .travis.yml]: Yes, That is fine.

File: .travis.yml (hunk 4 above)
Comment (Nyholm) [reply .travis.yml]: Yes, see https://github.com/symfony/symfony-docs/pull/8701#discussion_r152016042

File: .travis.yml (hunk 3 above)
Comment (Nyholm) [reply .travis.yml]: Thank you

File: composer.json (hunk 1 above)
Comment (Nyholm) [reply composer.json]: Hm.. I maybe got confused by this PR https://github.com/symfony/symfony/pull/23952

Comment (xabbuh) [reply composer.json]: That was merged into the `3.3` branch (I forgot to update the changelog in the PR but that was fixed after merging). `^3.3 || ^4.0` should be good here.

**Comment by Nyholm** [comment]
Thank you. I did not know you could

File: .travis.yml (hunk 4 above)
Comment (fbourigault) [reply .travis.yml]: So maybe a comment like *Test latest PHP version with lowest dependencies* would be more accurate.

File: .travis.yml [hunk 5]
```diff
@@ -6,16 +6,10 @@ cache:
...
-    - 5.5
-    - 5.6
-    - 7.0
-    - 7.1
-    - 7.2
-
 env:
     global:
         - TEST_COMMAND="composer test"
+        - SYMFONY_PHPUNIT_VERSION="6.3"
```
Comment (xabbuh) [inline .travis.yml]: Did tests fail without exposing this environment variable? The bridge should pick the best PHPUnit version automatically if you don't do it.

**Comment by fbourigault** [comment]
I restarted a job because it failed with a disk quota exceeded message. Build is now green.

File: .travis.yml [hunk 6]
```diff
@@ -24,26 +18,48 @@ branches:
 matrix:
     fast_finish: true
     include:
+          # Minimum supported Symfony version and latest PHP version
+        - php: 7.1
+          env: DEPENDENCIES="minimum" COVERAGE=true TEST_COMMAND="composer test-ci" SYMFONY_DEPRECATIONS_HELPER="weak" SYMFONY_PHPUNIT_VERSION="5.7"
+
+          # Test the latest stable release
         - php: 5.5
```
Comment (xabbuh) [inline .travis.yml]: Not sure you need this job. There are two other jobs making use of PHP 5.5.

**Comment by Nyholm** [comment]
Thank you for the review

File: .travis.yml (hunk 5 above)
Comment (Nyholm) [reply .travis.yml]: Yes, it did. It was some issues with mattiasnoback/config-test.

Comment (fbourigault) [reply .travis.yml]: Have you more details on those issues? I could look at it and maybe fix those in the library and so remove those variables from our travis file.

File: .travis.yml (hunk 6 above)
Comment (Nyholm) [reply .travis.yml]: Hm, maybe you are correct.

**Pull request merged**